artnet/ksy/artrdm.ksy

```diff
@@ -0,0 +1,86 @@
+# Development Notes
+
+meta:
+  id: artrdm
+  title: ArtRdm
+
+doc: |
+  The ArtRdm packet is used to transport all non-discovery RDM messages over Art-Net
+
+  Controller: 
+    Receive: No Action
+    Unicase Transmit: Yes
+    Broadcast: Not Allowed
+
+  Node Output Gateway:
+    Receive: No Action
+    Unicast Transmit: Yes
+    Broadcast: Not Allowed
+    
+  Node Input Gateway:
+    Receive: No Action
+    Unicast Transmit: Yes
+    Broadcast: Not Allowed
+
+  Media Server:
+    Receive: No Action
+    Unicast Transmit: Yes
+    Broadcast: Not Allowed
+
+seq:
+  - id: id
+    contents: [0x41, 0x72, 0x74, 0x2d, 0x4e, 0x65, 0x74, 0x00]
+    doc: | 
+      Array of 8 characters, the final character is a null termination.
+      Value = ‘A’ ‘r’ ‘t’ ‘-‘ ‘N’ ‘e’ ‘t’ 0x00
+    
+  - id: opcode
+    contents: [0x00, 0x83]
+    doc: OpIpProgReply (transmitted little endian)
+
+  - id: proto_version
+    contents: [0x00, 0x0e]
+    doc: Protocol version (constant 14, transmitted big endian)
+    
+  - id: rdm_version
+    type: u1
+    doc: Version of RDM specification supported by the device
+    enum: rdm_versions
+    
+  - size: 1
+    doc: Pad length to match ArtPoll
+    
+  - size: 7
+    doc: Spare transmitted as zero, receivers don't test
+    
+  - id: net
+    type: u1
+    doc: The top 7 bits of the Port-Address that should action this command
+    
+  - id: command
+    type: u1
+    doc: Defines the packet action
+    enum: command
+  
+  - id: address
+    type: u1
+    doc: The low 8 bits of the Port-Address that should action this packet
+    
+  - id: rdm_packet
+    size: 256
+    doc: The RDM packet data excluding the DMX StartCode. The maximum packet length is 255 + 2-byte checksum - 1-byte start code = 256 bytes
```

Comment:
@minorDeveloper

Reply:
The specification is wonderfully unhelpful. I don't think we'll be certain about the ranges of this packet until I've written the RDM spec itself. For now I'd suggest using `size-eos: true` and `max: 256` to give the following

```
  - id: rdm_packet
    size-eos: true
    doc: The RDM packet data excluding the DMX StartCode...
    valid:
      max: 256
```

If you're happy with this then feel free to resolve

Reply:
I will need to have a look at this to check that the `valid` block can actually work that way and constrain length. I want to also check a bit more on how `error-eos` works because there's a chance that can suppress errors if it hits the end of the stream unexpectedly but that might only be for delimited structures.